network/peer-monitoring-service/client/src/lib.rs

```diff
@@ -151,6 +156,51 @@ async fn start_peer_monitor_with_state(
     }
 }
 
+/// Creates a new peer state for peers that don't yet have one
+fn create_states_for_new_peers(
+    node_config: &NodeConfig,
+    peer_monitor_state: &PeerMonitorState,
+    time_service: &TimeService,
+    connected_peers_and_metadata: &HashMap<PeerNetworkId, PeerMetadata>,
+) {
+    for peer_network_id in connected_peers_and_metadata.keys() {
+        let state_exists = peer_monitor_state
+            .peer_states
+            .read()
+            .contains_key(peer_network_id);
+        if !state_exists {
+            peer_monitor_state.peer_states.write().insert(
+                *peer_network_id,
+                PeerState::new(node_config.clone(), time_service.clone()),
```

Comment:
Should `time_service` and `node_config` be `Arc<...>` if we're cloning it between every peer state, that's a lot of overhead.  Especially since they will be exactly the same.

Reply:
Yeah, the `time_service` is super cheap (i.e., it's a super lightweight struct, designed for cloning). Regarding `node_config`, I see your point, but given that we'll likely only have <= 100 copies (in the most extreme cases), I'm not super worried for now. We can always optimize it later.

Generally though, I agree. There's a lot of data items that we tend to clone and pass around (configs being one of them). I'll have a think about a better way of managing this for future improvements.

Reply:
Yeah, let's put a GitHub issue for removing this cloning of `node_config`,  I suspect it's not something you want to be  necessarily copying per node.  Especially since it has so many nested pieces.